File: python/emfmm3d_example.py
#!/usr/bin/env python

import fmm3dpy as fmm
import numpy as np


#
#  This is a sample code to demonstrate how to use
#  the fmm libraries
#
#
# Source to target, magnetic current only, output field example
#
n = 20000
nd = 1
sources = np.random.uniform(0,1,(3,n))

nt = 4000
targ = np.random.uniform(0,1,(3,nt))
eps = 10**(-5)

zk = 1.1 + 1j*0
h_current = np.random.uniform(0,1,(3,n)) + 1j*np.random.uniform(0,1,(3,n))
out = fmm.emfmm3d(eps=eps,zk=zk, sources=sources,h_current=h_current,targets=targ,ifE = 1)

#
# Example 2: source to target, electric current + charge, multiple densities, field + curl
#
#

nd = 3
zk = 1.1 + 1j*0
e_charge = np.random.uniform(0,1,(nd,n)) + 1j*np.random.uniform(0,1,(nd,n))
e_current = np.random.uniform(0,1,(nd,3,n)) + 1j*np.random.uniform(0,1,(nd,3,n))
out = fmm.emfmm3d(eps=eps,zk=zk, sources=sources,e_charge=e_charge,\
    e_current=e_current,targets=targ,nd=nd,ifE = 1, ifcurlE = 1)

File: matlab/emfmm3d_example.m

%
%
disp("This is an example matlab driver for the maxwell fmm");
disp("On output the code prints the field in example1 and, field+curl in example2");
disp(" ");
disp(" ");
%%   source to target, magnetic current only, field example
%

clear srcinfo targ ns U eps ifE ifcurlE ifdivE zk;
disp("Example 1: source to source, magnetic current, field only");
disp(" ");
disp(" ");

ns = 20000;
srcinfo.sources = rand(3,ns);
srcinfo.h_current = rand(3,ns)+1i*rand(3,ns);

nt = 4000;
targ = rand(3,nt);

eps = 1e-5;
ifE = 1; ifcurlE = 0; ifdivE=0;
zk = complex(1.1);
U = emfmm3d(eps,zk,srcinfo,targ,ifE,ifcurlE,ifdivE);
disp("pot=");
disp(U.E(:,1:3));

clear srcinfo targ ns U eps ifE ifcurlE ifdivE zk;
%
%%    source to target, electric current + charge, field + curl example
%
disp(" ");
disp(" ");
disp("Example 2: source to target, electric current+charge, multiple densities, field+curl");
disp(" ");
disp(" ");

ns = 20000;
nt = 19990;

nd = 3;
srcinfo.nd = nd;

ifE = 1; ifcurlE = 1; ifdivE = 0;

srcinfo.sources = rand(3,ns);
srcinfo.e_charge = rand(nd,ns)+1i*rand(nd,ns);
srcinfo.e_current = rand(nd,3,ns)+1i*rand(nd,3,ns);

targ = rand(3,nt);
eps = 1e-5;
zk = complex(1.1);

U = emfmm3d(eps,zk,srcinfo,targ,ifE,ifcurlE,ifdivE);

disp("field=");
disp(U.E(:,:,1:3));

disp("curl=");
disp(U.curlE(:,:,1:3));
clear srcinfo ns U eps ifE ifcurlE ifdivE nt targ nd zk;
